source/common/http/http3/conn_pool.h

```diff
@@ -121,13 +129,23 @@ class Http3ConnPoolImpl : public FixedHttpConnPoolImpl {
   // the HTTP3 active client does.
   bool trackStreamCapacity() override { return false; }
 
+  void setConnectResultCallback(PoolConnectResultCallback& callback) {
+    connect_callback_ = std::reference_wrapper<PoolConnectResultCallback>(callback);
+  }
+
+protected:
+  void onConnected(Envoy::ConnectionPool::ActiveClient&) override;
+
 private:
+  friend class Http3ConnPoolImplPeer;
+
   // Store quic helpers which can be shared between connections and must live
   // beyond the lifetime of individual connections.
   std::unique_ptr<Quic::PersistentQuicInfoImpl> quic_info_;
+  absl::optional<std::reference_wrapper<PoolConnectResultCallback>> connect_callback_;
```

Comment:
nit: i think we can use Envoy's OptRef<PoolConnectResultCallback> here to simplify this. (And I think we can avoid the <functional> include as a result, but I might be wrong).